backend/errors/httperror_test.go

```diff
@@ -0,0 +1,64 @@
+package errors
+
+import (
+	"bytes"
+	"errors"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestHandleHTTPError(t *testing.T) {
+	tests := []struct {
+		testName       string
+		w              *httptest.ResponseRecorder
+		err            error
+		statusCode     int
+		userMessage    string
+		output         string
+		wrappedMessage string
+		res            bool
+	}{
+		{
+			testName:       "Nil",
+			w:              httptest.NewRecorder(),
+			err:            nil,
+			statusCode:     http.StatusOK,
+			userMessage:    "",
+			output:         "",
+			wrappedMessage: "",
+			res:            false,
+		},
+		{
+			testName:       "Server Error 500",
+			w:              httptest.NewRecorder(),
+			err:            errors.New("foo"),
+			statusCode:     http.StatusInternalServerError,
+			userMessage:    "bar",
+			output:         "Error: foo",
+			wrappedMessage: "{\"error\":\"bar\"}",
+			res:            true,
+		},
+	}
+	for _, test := range tests {
+		currentWriter := log.Writer()
+		var buf bytes.Buffer
+		log.SetOutput(&buf)
+		res := HandleHTTPError(test.w, test.err, test.statusCode, test.userMessage)
+		log.SetOutput(currentWriter)
+		if test.res != res {
+			t.Errorf("%s: expected %t, got %t", test.testName, test.res, res)
+		}
```

Comment:
@j4qfrost nitpicking here, we haven't added this to our style guide yet, but in accordance with [https://google.github.io/styleguide/go/decisions#got-before-want](the google style guide for go) we prefer `got` before `want` for tests. Can you please update these before merging? The main benefit we get from this is just consistency when reading the messages in the event of a test failure/regression.